test/e2e/basic_test.go

```diff
@@ -383,11 +411,11 @@ func testReconcileGatewayRoutesOnStartup(t *testing.T, data *TestData, isIPv6 bo
 			return false, err
 		}
 
-		if len(routes) < expectedRtNumMin {
+		if len(routes) < expectedRtNum {
 			// Not enough routes, keep trying
 			return false, nil
-		} else if len(routes) > expectedRtNumMax {
-			return false, fmt.Errorf("found too many gateway routes, expected %d but got %d", expectedRtNumMax, len(routes))
+		} else if len(routes) > expectedRtNum {
+			return false, fmt.Errorf("found too many gateway routes, expected %d but got %d", expectedRtNum, len(routes))
 		}
```

Comment:
shouldn't this be replaces with a simple `require.Len(t, routes, expectedRtNum)` assertion now, or a simple `if len(routes) != expectedRtNum` test if we should return an error?

Reply:
How about just like this:

```
	if len(routes) != expectedRtNum {
			return false, nil
		}
```

We should not return an error above. This is because PollUntilContextTimeout will exit immediately if an error is returned.